akka-typed/src/main/scala/akka/typed/Props.scala

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+
+import java.util.concurrent.Executor
+
+import akka.annotation.{ ApiMayChange, DoNotInherit, InternalApi }
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext
+import scala.reflect.ClassTag
+
+object Props {
+
+  /**
+   * Empty props instance, should seldom be needed in user code but can be useful as a default props where
+   * custom configuration of an actor is possible.
+   */
+  val empty: Props = EmptyProps
+}
+
+/**
+ * Data structure for describing an actor’s props details like which
+ * executor to run it on. For each type of setting (e.g. [[DispatcherSelector]]
+ * or [[MailboxCapacity]]) the FIRST occurrence is used when creating the
+ * actor; this means that adding configuration using the "with" methods
+ * overrides what was configured previously.
+ *
+ * Deliberately not sealed in order to emphasize future extensibility by the
+ * framework—this is not intended to be extended by user code.
+ *
+ */
+@DoNotInherit
+@ApiMayChange
+abstract class Props private[akka] () extends Product with Serializable {
+  /**
+   * Reference to the tail of this Props list.
+   *
+   * The `next` reference is here so that it can form an
+   * internally linked list. Traversal of this list stops when encountering the
+   * [[EmptyProps]] object.
+   *
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def next: Props
+
+  /**
+   * Create a copy of this Props node with its `next` reference
+   * replaced by the given object. <b>This does NOT append the given list
+   * of configuration nodes to the current list!</b>
+   *
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def withNext(next: Props): Props
```

Comment:
Should it be open, I was under the impression it is only about being for our own implementations (cluster for example)?